source/binding/Statements.cpp

```diff
@@ -995,6 +1033,32 @@ bool CaseStatement::verifyConstantImpl(EvalContext& context) const {
     return true;
 }
 
+void CaseStatement::serializeTo(ASTSerializer& serializer) const {
+    serializer.write("condition", toString(condition));
+    serializer.write("check", toString(check));
+    serializer.startArray("items");
+    for (auto const& item : items) {
+        serializer.startObject("item");
+
+        serializer.startArray("expressions");
+        for (auto const& ex : item.expressions) {
+            if (ex)
+                serializer.serialize(*ex);
+        }
+        serializer.endArray();
+
+        serializer.startObject("stmt");
```

Comment:
This is now wrong in the opposite way; you're putting the plain statement as an element of the object, but objects have properties, not undecorated elements (basically that's the difference in JSON between properties and arrays). You probably just want `serializer.write("stmt", *item.stmt);`